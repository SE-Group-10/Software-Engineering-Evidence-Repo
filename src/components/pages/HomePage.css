.homepageHeader {

    text-align: center;
    padding-top: 15vh;
    margin-left: 0;
    padding-left: 0;
    padding-right: 0;
    background-image: url("../../assets/homepage_background.jpg");
    background-size: cover;
    background-repeat: no-repeat;

    position: relative;
    height: 95vh;
    width: 100%;

    color: white;
}

.homepageCTA{
    margin-top: 10vh;
    padding-bottom: 5vh;
}


#navbar-search{
    display: 'block';
    margin-left: auto;
    margin-right: auto;
}


