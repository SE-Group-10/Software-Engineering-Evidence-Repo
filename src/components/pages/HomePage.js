import React from "react";
import { Container, Form, FormControl, Button } from "react-bootstrap";
import {Link} from "react-router-dom";

const HomePage = () => {
  return (
    <Container style={{
      textAlign: 'center',
      paddingTop: '10vh',
      width: '100vw',
      color: '#00994C'
    }}>
      <div className="homepageHeader">
        <h1> Welcome to SEER! </h1>
        <br />
        <h2> A repository for Software Engineers to search for articles, <br />
        journals or research papers for evidences to support their ideas.</h2>
      </div>

      <div className="homepageCTA" style={{ marginTop: '10vh'}}>
      <h2> What are you looking for? </h2>
      <Form inline id="navbar-search" style={{ display: 'block',
                                              marginLeft: 'auto',
                                              marginRight:'auto',
                                            }}>
        <FormControl type="text" placeholder="Search for articles, books or researches" className="mr-sm-2" style={{
          width: '30vw',
        }} />
        <Button variant="outline-success"> <Link to="/search-result"><img src={require("../../assets/icons/search_icon.svg")} id="search_icon" /> </Link></Button>
      </Form>
      </div>
    </Container>
  );
};

export default HomePage;
