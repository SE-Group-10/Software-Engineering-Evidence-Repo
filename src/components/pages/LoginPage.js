import React from "react";
import { Container } from "react-bootstrap";
import LoginForm from "../register-login-components/LoginForm";

const LoginPage = () => {
  return (
    <div>
      <LoginForm />
    </div>
  );
};

export default LoginPage;
