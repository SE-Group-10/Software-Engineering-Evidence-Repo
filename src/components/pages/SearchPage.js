import React from "react";
import { Container, Row, Col } from "react-bootstrap";
import SearchField from "../general-components/SearchField";
import NavBar2 from "../general-components/NavBar2";
import SearchFilters from "../search-result-components/SearchFilters";

const SearchPage = () => {
  return (
    <div>
      <NavBar2 />
      <div>
        <Row>
          <Col lg="2" >
          <SearchFilters />
          </Col>
          <Col lg="8">
          <SearchField />
          </Col>
        </Row>
      </div>
    </div>
  );
};

export default SearchPage;
