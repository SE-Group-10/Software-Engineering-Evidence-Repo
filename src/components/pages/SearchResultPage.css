.searchSectionStyle {
  display: inline-block;
  padding-right: 5%;
}
.searchFilterResult {
  padding: 0;
}
.searchTableHeader {
  margin-top: 50px;
  color: #00994c;
  margin-bottom: 20px;
  text-align: center;
}

.searchTableStyles thead {
  text-align: center;
  overflow: auto;
  white-space: nowrap;
}

.searchTableStyles thead th {
  cursor: pointer;
}
.searchTableStyles thead th:hover {
  color: #00994c;
}

.searchTableStyles td {
  max-width: 300px;
  white-space: nowrap;
  overflow: auto;
  text-overflow: inherit;
  text-align: center;
  vertical-align: middle;
}
