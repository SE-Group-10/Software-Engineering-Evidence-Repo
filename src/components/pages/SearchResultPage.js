import React from "react";
import SearchFilters from "../search-result-components/SearchFilters";

const SearchResultPage = () => {
  return (
    <div>  
    <h1 style={{ textAlign:'center'}}> 
    Query Results: </h1>
    <SearchFilters />
    </div>
  );
};

export default SearchResultPage;
