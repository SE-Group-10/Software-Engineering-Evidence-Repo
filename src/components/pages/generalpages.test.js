// Required packages for testing
import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

// Components to be tested
import HomePage from './HomePage.js';
import LoginPage from './LoginPage.js';
import SearchPage from './SearchPage.js';
import SearchResultPage from './SearchResultPage.js';
import SignUpPage from './SignUpPage.js';



// Snapshot tests for UI testing.
test('Home Page Renders properly', () => {
    const tree = renderer.create(<Router><HomePage /></Router>).toJSON();
    expect(tree).toMatchSnapshot();
});

test('Login Page Renders properly', () => {
    const tree = renderer.create(<Router><LoginPage /></Router>).toJSON();
    expect(tree).toMatchSnapshot();
});

test('Search Page Renders properly', () => {
    const tree = renderer.create(<Router><SearchPage /></Router>).toJSON();
    expect(tree).toMatchSnapshot();
});

test('Search Result Page Renders properly', () => {
    const tree = renderer.create(<Router><SearchResultPage /></Router>).toJSON();
    expect(tree).toMatchSnapshot();
});

test('Sign Up Renders properly', () => {
    const tree = renderer.create(<Router><SignUpPage /></Router>).toJSON();
    expect(tree).toMatchSnapshot();
});



