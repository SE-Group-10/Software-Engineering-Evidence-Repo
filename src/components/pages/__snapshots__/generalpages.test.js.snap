// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page Renders properly 1`] = `
<div
  className="container"
  style={
    Object {
      "color": "#00994C",
      "paddingTop": "10vh",
      "textAlign": "center",
      "width": "100vw",
    }
  }
>
  <div
    className="homepageHeader"
  >
    <h1>
       Welcome to SEER! 
    </h1>
    <br />
    <h2>
       A repository for Software Engineers to search for articles, 
      <br />
      journals or research papers for evidences to support their ideas.
    </h2>
  </div>
  <div
    className="homepageCTA"
    style={
      Object {
        "marginTop": "10vh",
      }
    }
  >
    <h2>
       What are you looking for? 
    </h2>
    <form
      className="form-inline"
      id="navbar-search"
      style={
        Object {
          "display": "block",
          "marginLeft": "auto",
          "marginRight": "auto",
        }
      }
    >
      <input
        className="mr-sm-2 form-control"
        placeholder="Search for articles, books or researches"
        style={
          Object {
            "width": "30vw",
          }
        }
        type="text"
      />
      <button
        className="btn btn-outline-success"
        disabled={false}
        type="button"
      >
         
        <a
          href="/search-result"
          onClick={[Function]}
        >
          <img
            id="search_icon"
            src={
              Object {
                "ReactComponent": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                },
                "__esModule": true,
                "default": "search_icon.svg",
              }
            }
          />
           
        </a>
      </button>
    </form>
  </div>
</div>
`;

exports[`Login Page Renders properly 1`] = `
<div
  className="container"
>
  <div
    className="container"
  >
    <form
      className=""
    >
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="formBasicEmail"
        >
          Email address
        </label>
        <input
          className="form-control"
          id="formBasicEmail"
          placeholder="Enter email"
          type="email"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="formBasicPassword"
        >
          Password
        </label>
        <input
          className="form-control"
          id="formBasicPassword"
          placeholder="Password"
          type="password"
        />
      </div>
      <div
        className="form-group"
      >
        <div
          className="form-check"
        >
          <input
            className="form-check-input"
            disabled={false}
            id="formBasicCheckbox"
            type="checkbox"
          />
          <label
            className="form-check-label"
            htmlFor="formBasicCheckbox"
            title=""
          >
            Remember Me
          </label>
        </div>
      </div>
      <button
        className="btn btn-primary"
        disabled={false}
        type="submit"
      >
        Login
      </button>
    </form>
  </div>
</div>
`;

exports[`Search Page Renders properly 1`] = `
<div
  className="container"
>
  <div
    className="searchFieldContainer container-fluid"
  >
    <div
      className="container"
    >
      <form
        className=""
      >
        <div
          className="form-group"
        >
          <label
            className="form-label"
            htmlFor="descriptionSearch"
          >
            <h4>
              Description
            </h4>
          </label>
          <input
            className="form-control"
            id="descriptionSearch"
            type="text"
          />
        </div>
        <div
          className="form-group"
        >
          <div
            className="row"
          >
            <div
              className="col-sm-2"
            >
              <label
                className="form-label"
                htmlFor="dateRangeSearch"
              >
                <h4>
                  Date Range:
                </h4>
              </label>
            </div>
            <div
              className="col"
            >
              <input
                className="form-control"
                id="dateRangeSearch"
                type="date"
              />
            </div>
            <div
              className="col-sm-1"
            >
              <h4>
                To
              </h4>
            </div>
            <div
              className="col"
            >
              <input
                className="form-control"
                id="dateRangeSearch"
                type="date"
              />
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-sm-1"
          >
            <h4>
              If
            </h4>
          </div>
          <div
            className="col"
          >
            <select
              className="form-control"
            >
              <option
                value="0"
              >
                Name of Field
              </option>
              <option
                value="article_title"
              >
                Article Title
              </option>
              <option
                value="article_name"
              >
                Article Name
              </option>
              <option
                value="article_author"
              >
                Author
              </option>
            </select>
          </div>
          <div
            className="col-sm-2"
          >
            <select
              className="form-control"
            >
              <option
                value="0"
              >
                Operator
              </option>
              <option
                value="contains"
              >
                Contains
              </option>
              <option
                value="not_contains"
              >
                Does Not Contain
              </option>
              <option
                value="begins_with"
              >
                Begins With
              </option>
              <option
                value="equals"
              >
                Is Equal To
              </option>
              <option
                value="less_than"
              >
                Is Less Than
              </option>
              <option
                value="less_than_equals"
              >
                Is Less Than Or Equal To
              </option>
              <option
                value="more_than"
              >
                Is More Than
              </option>
              <option
                value="more_than_equals"
              >
                Is More Than Or Equal To
              </option>
            </select>
          </div>
          <div
            className="col"
          >
            <select
              className="form-control"
            >
              <option>
                Value
              </option>
            </select>
          </div>
          <div
            className="col-sm-1"
          >
            <a>
              <img
                alt="add-search"
                src={
                  Object {
                    "ReactComponent": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "__esModule": true,
                    "default": "plus_icon.svg",
                  }
                }
              />
            </a>
          </div>
          <div
            className="col-sm-1"
          >
            <a>
              <img
                alt="add-search"
                src={
                  Object {
                    "ReactComponent": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "__esModule": true,
                    "default": "dash_icon.svg",
                  }
                }
              />
            </a>
          </div>
        </div>
        <button
          className="btn btn-primary"
          disabled={false}
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`Search Result Page Renders properly 1`] = `
<div>
  <h1
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Query Results: 
  </h1>
  <div
    className="pro-sidebar"
    style={
      Object {
        "width": undefined,
      }
    }
  >
    <div
      className="pro-sidebar-inner"
    >
      <div
        className="pro-sidebar-layout"
      >
        <div
          className="pro-sidebar-header"
        >
          <nav
            className="pro-menu shaped square"
          >
            <ul>
              <li
                className="pro-menu-item pro-sub-menu"
              >
                <div
                  className="pro-inner-item"
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  <span
                    className="pro-item-content"
                  >
                    Search Filters:
                  </span>
                  <span
                    className="pro-arrow-wrapper"
                  >
                    <span
                      className="pro-arrow"
                    />
                  </span>
                </div>
                <div
                  className="react-slidedown pro-inner-list-item"
                  onTransitionEnd={[Function]}
                >
                  <div>
                    <ul>
                      <li
                        className="pro-menu-item"
                      >
                        <div
                          className="pro-inner-item"
                          role="button"
                          tabIndex={0}
                        >
                          <span
                            className="pro-item-content"
                          >
                            Component 1
                          </span>
                        </div>
                      </li>
                      <li
                        className="pro-menu-item pro-sub-menu"
                      >
                        <div
                          className="pro-inner-item"
                          onClick={[Function]}
                          onKeyPress={[Function]}
                          role="button"
                          tabIndex={0}
                        >
                          <span
                            className="pro-item-content"
                          >
                            Sub Component 1
                          </span>
                          <span
                            className="pro-arrow-wrapper"
                          >
                            <span
                              className="pro-arrow"
                            />
                          </span>
                        </div>
                        <div
                          className="react-slidedown pro-inner-list-item"
                          onTransitionEnd={[Function]}
                        >
                          <div>
                            <ul />
                          </div>
                        </div>
                      </li>
                    </ul>
                  </div>
                </div>
              </li>
            </ul>
          </nav>
        </div>
        <div
          className="pro-sidebar-content"
        >
          <nav
            className="pro-menu shaped square"
          >
            <ul>
              <li
                className="pro-menu-item pro-sub-menu"
              >
                <div
                  className="pro-inner-item"
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  <span
                    className="pro-item-content"
                  >
                    Sort By:
                  </span>
                  <span
                    className="pro-arrow-wrapper"
                  >
                    <span
                      className="pro-arrow"
                    />
                  </span>
                </div>
                <div
                  className="react-slidedown pro-inner-list-item"
                  onTransitionEnd={[Function]}
                >
                  <div>
                    <ul>
                      <li
                        className="pro-menu-item"
                      >
                        <div
                          className="pro-inner-item"
                          role="button"
                          tabIndex={0}
                        >
                          <span
                            className="pro-item-content"
                          >
                            Component 1
                          </span>
                        </div>
                      </li>
                      <li
                        className="pro-menu-item pro-sub-menu"
                      >
                        <div
                          className="pro-inner-item"
                          onClick={[Function]}
                          onKeyPress={[Function]}
                          role="button"
                          tabIndex={0}
                        >
                          <span
                            className="pro-item-content"
                          >
                            Sub Component 1
                          </span>
                          <span
                            className="pro-arrow-wrapper"
                          >
                            <span
                              className="pro-arrow"
                            />
                          </span>
                        </div>
                        <div
                          className="react-slidedown pro-inner-list-item"
                          onTransitionEnd={[Function]}
                        >
                          <div>
                            <ul />
                          </div>
                        </div>
                      </li>
                    </ul>
                  </div>
                </div>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </div>
    <div
      aria-label="overlay"
      className="overlay"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    />
  </div>
</div>
`;

exports[`Sign Up Renders properly 1`] = `
<div
  className="container"
>
  <div
    className="container"
  >
    <h1
      style={
        Object {
          "color": "#00994C",
          "textAlign": "center",
        }
      }
    >
       Register with SEER
    </h1>
    <form
      className=""
    >
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="formBasicFirstName"
        >
          First Name
        </label>
        <input
          className="form-control"
          id="formBasicFirstName"
          placeholder="John"
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="formBasicLastName"
        >
          Last Name
        </label>
        <input
          className="form-control"
          id="formBasicLastName"
          placeholder="Doe"
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="formUsername"
        >
          Username
        </label>
        <input
          className="form-control"
          id="formUsername"
          placeholder="Enter Username"
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="formBasicEmail"
        >
          Email address
        </label>
        <input
          className="form-control"
          id="formBasicEmail"
          placeholder="john.doe@gmail.com"
          type="email"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="formBasicPassword"
        >
          Password
        </label>
        <input
          className="form-control"
          id="formBasicPassword"
          placeholder="Password must be 8 or more characters"
          type="password"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="formBasicEmail"
        >
          Affilitiation
        </label>
        <input
          className="form-control"
          id="formBasicEmail"
          placeholder="Enter your company, school or university"
          type="text"
        />
      </div>
      <label
        className="form-label"
      >
        Select your account type:
      </label>
      <div
        className="form-group"
      >
        <div
          className="form-check"
        >
          <input
            className="form-check-input"
            disabled={false}
            id="formBasicRadiobox"
            name="account_type"
            type="radio"
          />
          <label
            className="form-check-label"
            htmlFor="formBasicRadiobox"
            title=""
          >
            Standard User
          </label>
        </div>
        <div
          className="form-check"
        >
          <input
            className="form-check-input"
            disabled={false}
            id="formBasicRadiobox"
            name="account_type"
            type="radio"
          />
          <label
            className="form-check-label"
            htmlFor="formBasicRadiobox"
            title=""
          >
            Analyst (SERL Members only)
          </label>
        </div>
        <div
          className="form-check"
        >
          <input
            className="form-check-input"
            disabled={false}
            id="formBasicRadiobox"
            name="account_type"
            type="radio"
          />
          <label
            className="form-check-label"
            htmlFor="formBasicRadiobox"
            title=""
          >
            Moderator (requires authentication)
          </label>
        </div>
        <div
          className="form-check"
        >
          <input
            className="form-check-input"
            disabled={false}
            id="formBasicRadiobox"
            name="account_type"
            type="radio"
          />
          <label
            className="form-check-label"
            htmlFor="formBasicRadiobox"
            title=""
          >
            Administrator (requires authentication)
          </label>
        </div>
      </div>
      <button
        className="btn btn-primary"
        disabled={false}
        type="submit"
      >
        Register
      </button>
    </form>
  </div>
</div>
`;
